### PR TITLE
Migrating to Swift Package Manager

### PR DESCRIPTION
**Pull request by nontapat-tw**
Due to the outdated dependencies version and the issues with [unsupported Xcode 14](https://github.com/techops-recsys-lateral-hiring/mobiledev-wechatmoments-swift/pull/9), extra steps are needed from the interviewee before they can start their actual tasks. 

So, switching from **Cocoapods** to **Swift Package Manager** can solve these issues and let them focus on the interview.

**Tasks**
- [x] deintegrate **Cocoapods**
- [x] remove `xcworkspace` and other files associated with **Cocoapods**
- [x] integrate Swift Package Manager and make necessary code changes
- [x] update gitignore to include `.DS_Store`
- [x] update `README.md`

### CODE REVIEW

**Review by siroratt-tw — APPROVED** [review]
I have confirmed the modifications in the pull request on my end (using Intel 2019), and the project runs flawlessly as intended. 
Screenshot 2567-01-07 at 10 36 58

**Comment by siroratt-tw** [comment]
I have one more thing to mention. @nontapat-tw, could you assist in verifying that the two commands mentioned in the readme below are run perfectly fine in your end?

Execute npm install -g mountebank
Execute mb --configfile imposters.ejs

File: README.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 ### Setup the project
 1. Locate the current directory in terminal
-2. Execute `pod install` (or `sudo arch -x86_64 gem install ffi && arch -x86_64 pod install` for Apple M1)
-3. Execute `npm install -g mountebank`
-4. Execute `mb --configfile imposters.ejs`
-5. Open `WeChatMoments.xcworkspace` in the current directory
-6. Run the project and make sure the app can start correctly(don't worry about if the features are working properly)
+2. Execute `npm install -g mountebank`
+3. Execute `mb --configfile imposters.ejs`
+4. Open `WeChatMoments.workspace` in the current directory and ensure that SPM (Swift Package Manager) frameworks are ready
```
Comment (shebin-tw) [inline README.md]: please change `WeChatMoments.workspace` to `WeChatMoments.xcodeproj`

**Comment by shebin-tw** [comment]
@nontapat-tw  Tested in **Xcode 15.0 - M1 Mac**. working as expected 
Screenshot 2024-01-18 at 10 05 54 AM

Also, the below commands are working for me.
`npm install -g mountebank`
`mb --configfile imposters.ejs`

**Comment by nontapat-tw** [comment]
Thanks @siroratt-tw and @shebin-tw for help verifying this. I've tested on **M1** and **Intel** macbooks (both are using Xcode 15.0) and the commands works fine on my end.

@siroratt-tw I'm not sure if the error you encountered is because of different node version. Could you provide me the error logs?

**Comment by siroratt-tw** [comment]
Hi all,

after i did `brew upgrade and clean up`  the error is resolved. so @shebin-tw. if this PR looks good for you please help us merge. thx

File: README.md (hunk 1 above)
Comment (nontapat-tw) [reply README.md]: nice catch!

**Review by shebin-tw — APPROVED** [review]

**Comment by shebin-tw** [comment]
@wolesolana can you please help here to merge this PR. I don't have access to merge.

**Comment by wolesolana** [comment]
> @wolesolana can you please help here to merge this PR. I don't have access to merge.

@shebin-tw unfortunately, I don't have write access to this repo. Maybe @xguo-tw can help us out?

**Pull request merged**